service/src/main/java/bio/terra/cbas/service/RunSetsService.java

```diff
@@ -243,28 +244,32 @@ public void triggerWorkflowSubmission(
         methodVersion,
         runSetStateResponse.successfullyInitializedWorkflowIds(),
         userToken);
+
+    capturePostSubmitMetrics(runSet.runSetId(), runStateResponseList);
   }
 
   private WdsRecordResponseDetails fetchWdsRecords(
       WdsService wdsService, RunSetRequest request, RunSet runSet, BearerToken userToken) {
+    Timer.Sample wdsFetchRecordsSample = micrometerMetrics.startTimer();
+
     String recordType = request.getWdsRecords().getRecordType();
+    int totalRecords = request.getWdsRecords().getRecordIds().size();
 
     WdsRecordResponseDetails responseDetails =
         wdsService.getRecords(recordType, request.getWdsRecords().getRecordIds(), userToken);
     Map<String, String> recordIdsWithError = responseDetails.recordIdsWithError();
-    Timer.Sample wdsFetchRecordsSample = micrometerMetrics.startTimer();
+
     micrometerMetrics.stopTimer(
         wdsFetchRecordsSample,
         "wds_fetch_records_timer",
         RunSet.RUN_SET_ID_COL,
         runSet.runSetId().toString(),
+        "total_records_requests",
+        String.valueOf(totalRecords),
```

Comment:
I added this here as it would be useful to know for how much time it took to fetch WDS records for N records.